View cell rendering

richardolsson opened the pull request:
## Description
This PR implements custom rendering for View cells of all column types.

## Screenshots
![image](https://user-images.githubusercontent.com/550212/147276544-f21433c5-62ae-4f00-b8d5-c5dbf60cb6d4.png)

## Changes
* Use default rendering for `person_field` columns
* Use standard boolean rendering for `person_tag`, `person_query` and `local_bool` columns
* Use custom components for other column types
* Create utility function to generate MUI Grid column definitions from Zetkin view column objects
* Test utility and all custom cell components

## Notes to reviewer
I've edited View ID 2 ([link to localhost](http://localhost:3000/organize/1/people/views/2)) on the dev server to contain columns of all types.

Note that resizing the columns affect how they rendered. Most just show an ellipsis, but `local_person` also hides the name entirely when less than 100px wide.

## Related issues
Resolves #487

djbusstop commented on src/components/views/ViewDataTable/utils.tsx:
```diff
@@ -0,0 +1,63 @@
+    return parseInt(nameFields[1]);
+}
+
+export function makeGridColDef(viewCol: ZetkinViewColumn, orgId: number | string) : GridColDef {
```
Would love a comment here that said something like "maps a `ZetkinViewColumn` to an MUI DataGrid `GridColDef` to show the correct content." Just because I love comments.

djbusstop commented on src/components/views/ViewDataTable/cells/PersonNotesViewCell.tsx:
```diff
@@ -0,0 +1,37 @@
+    created: string;
+    id: number;
+    text: string;
+}[] | null>;
```
Why have the params include the possible null type? I would put `params` as an optional prop? Maybe this will make sense to me elsewhere, but is `null` a valid value to be considered "params"

djbusstop replied:
Left a previous comment but I realise now this means the VALUE is null.

richardolsson commented:
I've added some comments, fixed a bug related to dates that I hadn't seen before (might be because it's new years eve) and merged `main` to fix a conflict.
